src/form-builder/DateField.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { useField, FieldHookConfig } from 'formik';
+
+import { FieldProps } from './types';
+import { getMessage } from '../utils/i18n';
+import {
+  ValidationFunction,
+  chainValidations,
+  required,
+} from '../utils/validation';
+
+import { SimpleDate } from '@department-of-veterans-affairs/component-library/Date';
+
+export type ComponentLibraryDatePart = {
+  value: string;
+  dirty: boolean;
+};
+
+/**
+ * What the Date in the component library expects.
+ */
+export type ComponentLibraryDateValue = {
+  day: ComponentLibraryDatePart;
+  month: ComponentLibraryDatePart;
+  year: ComponentLibraryDatePart;
+};
+
+export type DateProps = FieldProps<string>;
+
+/**
+ * Convert the date string passed to DateField to the object expected by the
+ * component library Date component.
+ */
+const dateStringToValue = (dateString = '') => {
+  // TODO: Make this conversion handle edge cases
+  const [month = '', day = '', year = ''] = dateString.split('-');
+  return {
+    day: { value: day, dirty: false },
+    month: { value: month, dirty: false },
+    year: { value: year, dirty: false },
+  };
+};
+
+const dateValueToString = ({ day, month, year }: ComponentLibraryDateValue) => {
+  return `${month.value}-${day.value}-${year.value}`;
+};
+
+/**
+ * Super basic date string validation. Only checks for the non-empty strings
+ * between the `-`s.
+ */
+const validDate: ValidationFunction<string> = (
+  value: string,
+  props: DateProps
+) => {
+  if (props.required) {
+    const [month = '', day = '', year = ''] = value.split('-');
+    if (!month || !day || !year) {
+      return typeof props.required === 'string'
+        ? props.required
+        : getMessage('required.default');
+    }
+  }
```

Comment:
Should `validDate` be concerned about `required`? `chainValidations` allows those validations to be treated independently, so I wonder if the individual functions should do the same.

Reply:
Maybe not. What I was thinking was that we should only verify the date is valid if the date is also required. Like, for dates that aren't required, we don't need to make sure they're complete.

But maybe that doesn't make sense after all. Without the required stipulation, it becomes "If you enter this optional date, be sure it's complete," which sounds fine. :shrug:

Reply:
Yeah, I think it makes sense to still do some basic "is this a date" validation even if the date isn't required.